### DISCUSSION TITLE
请问github的回调地址填什么呀？

### DISCUSSION THREAD

**Discussion by ma3252788** [AlistGo/alist#2831] [General]
已经解决。

添加教程：https://www.machunjie.com/setup/1362.html

**Comment by Suzy-Rose** [comment]
> 已经解决。

请问怎么填写，一直没弄好

**Reply by ma3252788** [reply]
回调地址：https://your.domain.com/api/auth/github_callback

添加教程：[https://www.machunjie.com/setup/1362.html](https://www.machunjie.com/setup/1362.html)

**Reply by erbanku** [reply]
> 回调地址：[your.domain.com/api/auth/github_callback](https://your.domain.com/api/auth/github_callback)
> 
> 添加教程：[machunjie.com/setup/1362.html](https://www.machunjie.com/setup/1362.html)
@ma3252788 
**请问怎么解决的呀?**
**我这边按上述教程操作无法解决。**
#### 一些信息
AlistV3首页：`https://v3.domain.com/@login`
我的`GitHub OAuth app`配置：
GitHub APP：
Homepage URL: `https://v3.domain.com/@login`
Callback URL: `https://v3.domain.com/api/auth/github_callback`
#### 错误信息
```
https://v3.domain.com/api/auth/github_callback?error=redirect_uri_mismatch&error_description=The+redirect_uri+MUST+match+the+registered+callback+URL+for+this+application.&error_uri=https%3A%2F%2Fdocs.github.com%2Fapps%2Fmanaging-oauth-apps%2Ftroubleshooting-authorization-request-errors%2F%23redirect-uri-mismatch
```

**Reply by Suzy-Rose** [reply]
Homepage URL: https://v3.domain.com/ 这样填

**Reply by Suzy-Rose** [reply]
写你自己网站的域名

**Reply by erbanku** [reply]
> 写你自己网站的域名
> 
> 

我那样填写也不行✋，错误信息相同。
感觉是Callback URL有问题🤨，可是我按照上面说的地址写的呀，不应该呀。

**Reply by Suzy-Rose** [reply]
> > 写你自己网站的域名
> 
> 我那样填写也不行✋，错误信息相同。 感觉是Callback URL有问题🤨，可是我按照上面说的地址写的呀，不应该呀。

Homepage URL: https://v3.domain.com/ 
Callback URL: http://v3.domain.com/api/auth/github_callback

**Comment by Debcharon** [comment]
無需切換至https，callback URL 填寫http，否則會出現：[Troubleshooting authorization request errors](https://docs.github.com/en/developers/apps/managing-oauth-apps/troubleshooting-authorization-request-errors)

**Reply by erbanku** [reply]
还是不行🥲

**Comment by evlic** [comment]
一样的问题，结合报错信息来看，github 会要求 callback 和 redirect url 相同，但是 alist 会将当前 url 传给 github，以期验证完成时恢复当前访问。

**Reply by erbanku** [reply]
> 一样的问题，结合报错信息来看，github 会要求 callback 和 redirect url 相同，但是 alist 会将当前 url 传给 github，以期验证完成时恢复当前访问。

我这边排查发现是Nginx问题，按照[此文档](https://alist.nn.ci/guide/install/reverse-proxy.html)修改就好了。

**Reply by evlic** [reply]
是的……不过我通过读源码把问题找出来了hhh

```conf
nginx 里添加
        proxy_set_header X-Forwarded-Proto $http_x_forwarded_proto;
```

相关源代码 server/common/base.go:12
```go
func GetApiUrl(r *http.Request) string {
	api := conf.Conf.SiteURL
	if strings.HasPrefix(api, "http") {
		return api
	}
	if r != nil && api == "" {
		protocol := "http"
		if r.TLS != nil || r.Header.Get("X-Forwarded-Proto") == "https" {
			protocol = "https"
		}
		host := r.Host
		if r.Header.Get("X-Forwarded-Host") != "" {
			host = r.Header.Get("X-Forwarded-Host")
		}
		api = fmt.Sprintf("%s://%s", protocol, stdpath.Join(host, api))
	}
	strings.TrimSuffix(api, "/")
	return api
}

```

**Reply by Silentely** [reply]
3.10.1后修改也还是出现The+redirect_uri+MUST+match+the+registered+callback+URL+for+this+application.&error_uri
